packages/ui/Measurable/Measurable.js

```diff
@@ -0,0 +1,88 @@
+/* eslint-disable react/sort-prop-types */
+
+/**
+ * A higher-order component that adds the ability to measure nodes conveniently.
+ *
+ * @module ui/Measurable
+ * @exports Measurable
+ */
+
+import hoc from '@enact/core/hoc';
+import React, {useState, useCallback} from 'react';
+
+/**
+ * Default config for {@link ui/Measurable.Measurable}.
+ *
+ * @memberof ui/Measurable.Measurable
+ * @hocconfig
+ */
+const defaultConfig = {
+	/**
+	 * Configures the prop name to pass the measurement object.
+	 *
+	 * @type {String}
+	 * @default 'measurement'
+	 * @memberof ui/Measurable.Measurable.defaultConfig
+	 */
+	measurementProp: 'measurement',
+
+	/**
+	 * Configures the prop name to pass that represents the element to measure.
+	 *
+	 * @type {String}
+	 * @default 'forwardRef'
+	 * @memberof ui/Measurable.Measurable.defaultConfig
+	 */
+	refProp: 'forwardRef'
+};
+
+function useMeasurable () {
+	const [measurement, setMeasurement] = useState();
+	const ref = useCallback(node => {
+		if (node != null) {
+			setMeasurement(node ? node.getBoundingClientRect() : null);
```

Comment:
Is this ternary or the `if` above redundant? Is using just this line better since it would invalidate the `measurement` state when the node were unmounted?

Reply:
@Djspaceg Also this

Reply:
So if `node` is null or undefined, `setMeasurement` won't be executed at all. If it's some other value, like false or something, then the measurement will be set to null. What may make this more meaningful would be changing `node` to the more detailed `node.getBoundingClientRect`. WDYT?

Reply:
`node` should either be a Node or `null`. if it's set to `null` because the `ref` is set to a Node that is unmounted, we should call `setMeasurement` with `null` to clear the measurements since they aren't accurate any longer. As I read the code now, `measurement` will be out of sync.